Tests/ML/augmentations/test_transform_pipeline.py

```diff
@@ -0,0 +1,164 @@
+#  ------------------------------------------------------------------------------------------
+#  Copyright (c) Microsoft Corporation. All rights reserved.
+#  Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+#  ------------------------------------------------------------------------------------------
+import random
+
+import PIL
+import pytest
+import torch
+from torchvision.transforms import CenterCrop, ColorJitter, RandomAffine, RandomErasing, RandomHorizontalFlip, \
+    RandomResizedCrop, Resize, ToTensor
+from torchvision.transforms.functional import to_tensor
+
+from InnerEye.Common.common_util import is_windows
+from InnerEye.ML.augmentations.image_transforms import AddGaussianNoise, ElasticTransform, ExpandChannels, RandomGamma
+from InnerEye.ML.augmentations.transform_pipeline import ImageTransformationPipeline, \
+    create_cxr_transforms_from_config
+
+# create_transform_pipeline_from_config
+from Tests.SSL.test_data_modules import cxr_augmentation_config
+
+import numpy as np
+
+
+@pytest.mark.parametrize("use_different_transformation_per_channel", [True, False])
+def test_torchvision_on_various_input(use_different_transformation_per_channel: bool) -> None:
+    """
+    This tests that we can run transformation pipeline with out of the box torchvision transforms on various types
+    of input: PIL image, 3D tensor, 4D tensors. Tests that use_different_transformation_per_channel has the correct
+    behavior.
+    """
+    image = np.ones([256, 256]) * 255.
+    image[100:150, 100:200] = 1
+    transform = ImageTransformationPipeline(
+        [CenterCrop(224),
+         RandomErasing(),
+         RandomAffine(degrees=(10, 12), shear=15, translate=(0.1, 0.3))
+         ],
+        use_different_transformation_per_channel)
+
+    # Test PIL image input
+    image = PIL.Image.fromarray(image).convert("L")
+    transform(image)
+
+    # Test image as [C, H. W] tensor
+    image_as_tensor = to_tensor(image)
+    transform(image_as_tensor)
+
+    # Test image as [1, 1, H, W]
+    image_as_tensor = image_as_tensor.unsqueeze(0)
+    assert image_as_tensor.shape == torch.Size([1, 1, 256, 256])
+    transformed = transform(image_as_tensor)
+    assert isinstance(transformed, torch.Tensor)
+    assert transformed.shape == torch.Size([1, 1, 224, 224])
+
+    # Test with a fake scan [C, Z, H, W] -> [25, 34, 256, 256]
+    test_4d_tensor = torch.ones([25, 34, 256, 256]) * 255.
+    test_4d_tensor[..., 100:150, 100:200] = 1
+    transformed = transform(test_4d_tensor)
+    assert isinstance(transformed, torch.Tensor)
+    assert transformed.shape == torch.Size([25, 34, 224, 224])
+
+    # Same transformation should be applied to all slices and channels.
+    assert torch.isclose(transformed[0, 0], transformed[1, 1]).all() != use_different_transformation_per_channel
+
+@pytest.mark.skipif(is_windows(), reason="OOM on windows")
```

Comment:
maybe you can make it run also on Windows if you reduce tensor sizes? Would also make the tests a bit faster